tests/unit/json_test.c

```diff
@@ -847,6 +847,78 @@ static void test_iterator_current(void)
     JsonDestroy(json);
 }
 
+static bool VisitPrimitive(JsonElement *primitive, void *data)
+{
+    assert_int_equal(JsonGetElementType(primitive), JSON_ELEMENT_TYPE_PRIMITIVE);
+
+    Writer *writer = data;
+    WriterWrite(writer, JsonPrimitiveGetAsString(primitive));
+    WriterWriteChar(writer, ',');
+    return true;
+}
+
+static bool VisitArray(JsonElement *array, void *data)
+{
+    assert_int_equal(JsonGetElementType(array), JSON_ELEMENT_TYPE_CONTAINER);
+    assert_int_equal(JsonGetContainerType(array), JSON_CONTAINER_TYPE_ARRAY);
+
+    Writer *writer = data;
+    WriterWriteF(writer, "[%zd]", JsonLength(array));
```

Comment:
```suggestion
    WriterWriteF(writer, "[%zd],", JsonLength(array));
```

Adding a comma here would make the trace more understandable, at least for me.

(Remember to also update expected trace below if you change this).

Reply:
I like how commas only separate values. `[2]value1,value2,` says those two values are in an array. But in general this is not at all important. A proper trace would have to be smarter.